packages/framework/tests/Unit/Component/Money/MoneyTest.php

```diff
@@ -0,0 +1,582 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrameworkBundle\Unit\Component\Money;
+
+use Iterator;
+use Litipk\BigNumbers\Errors\BigNumbersError;
+use PHPUnit\Framework\TestCase;
+use Shopsys\FrameworkBundle\Component\Money\Money;
+
+class MoneyTest extends TestCase
+{
+    /**
+     * @dataProvider fromStringProvider
+     * @param string $string
+     * @param string $expected
+     */
+    public function testFromString(string $string, string $expected): void
+    {
+        $money = Money::fromString($string);
+
+        $this->assertEquals($expected, $money->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function fromStringProvider(): Iterator
+    {
+        yield ['0', '0'];
+        yield ['-0', '-0'];
+        yield ['-0.0', '-0.0'];
+        yield ['1', '1'];
+        yield ['-1', '-1'];
```

Comment:
I wonder - can money be negative? Can You have negative amount of money? It sort of doesn't make sense for me.

Reply:
Well, it may represent a debt. Negative money amounts are commonly used imo.

Reply:
I have 200 crowns, I'll give you 300 crowns (subtraction). Will You have 300 and I -100 or this is impossible operation?

The answer is "Depends on domain". So what makes sense in our shop domain and why?

Reply:
I'm not sure here. In the current scope of the modification, I'm comfortable with keeping the behavior unchanged which means giving negative result (as would working with float|string).

Reply:
I'm really uncomfortable with such solution. We are coming with `Money` concept, not with `Number` concept. I need to understand our money domain before I can say whether negative money is valid. We should be serious when working with money. And I want to introduce as much restrictions as possible to avoid unintended behavior or errors.

Reply:
Let's keep negative money and document this decision.